plib/library/Rndc.php

```diff
@@ -11,66 +11,85 @@ private function _call(Modules_SlaveDnsManager_Slave $slave, $arguments, $verbos
         $output = implode("\n", $out);
-
+        
         if ($verbose) {
             if ($code != 0) {
                 throw new pm_Exception("$command $arguments\n$output\n\nError code: $code");
             }
             return $output;
         }
-
+        
         if ($code != 0) {
             // Cannot send output header due to possible API-RPC calls
             error_log("Error code $code: $output");
         }
-
+        
         return $code == 0;
     }
-
+    
     public function addZone($domain, Modules_SlaveDnsManager_Slave $slave = null)
     {
         $slaves = null === $slave ? Modules_SlaveDnsManager_Slave::getList() : [$slave];
         foreach ($slaves as $slave) {
-            $this->_call($slave, "addzone \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$slave->getRndcView()}\"" .
+            $multiView = $this->multipleView($slave->getRndcView());
+            foreach ($multiView as $view) {
+                $this->_call($slave, "addzone \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$view}\"" .
                 " \"{ type slave; file \\\"{$domain}\\\"; masters { {$slave->getMasterPublicIp()}; }; };\"");
+            }
         }
     }
-
+    
     public function updateZone($domain, Modules_SlaveDnsManager_Slave $slave = null)
     {
         $slaves = null === $slave ? Modules_SlaveDnsManager_Slave::getList() : [$slave];
         foreach ($slaves as $slave) {
-            $result = $this->_call($slave, "refresh \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$slave->getRndcView()}\"");
-            if (false === $result) {
-                $this->addZone($domain, $slave);
+            $multiView = $this->multipleView($slave->getRndcView());
+            foreach ($multiView as $view) {
+                $result = $this->_call($slave, "refresh \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$view}\"");
+                if (false === $result) {
+                    $this->addZone($domain, $slave);
+                }
             }
         }
     }
-
+    
     public function deleteZone($domain, Modules_SlaveDnsManager_Slave $slave = null)
     {
         $slaves = null === $slave ? Modules_SlaveDnsManager_Slave::getList() : [$slave];
         foreach ($slaves as $slave) {
             $slaveStatus = $this->checkStatus($slave);
             // version: 9.9.4-RedHat-9.9.4-51.el7_4.2 (none) <id:8f9657aa>
             // version: BIND 9.10.3-P4-Ubuntu <id:ebd72b3> (none)
-            $cleanFlag = (preg_match("/version: (BIND )?(9\.10\.\d+|9\.11\.\d+|9\.12\.\d+)/", $slaveStatus)) ? "-clean" : ""; 
-
-            $this->_call($slave, "delzone $cleanFlag \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$slave->getRndcView()}\"");
+            $cleanFlag = (preg_match("/version: (BIND )?(9\.10\.\d+|9\.11\.\d+|9\.12\.\d+)/", $slaveStatus)) ? "-clean" : "";
+            
+            $multiView = $this->multipleView($slave->getRndcView());
+            foreach ($multiView as $view) {
```

Comment:
According to the comment below, you can replace it with the `foreach ($slave->getRndcViews() as $view)`